portfolio/src/main/webapp/script.js

```diff
@@ -64,26 +64,52 @@ function filterCourseDisplay(courseType = 'course') {
  * Fetch comments from server and display on DOM.
  */
 function loadComments() {
-  const max = document.getElementById('max-num-comments').value;
-  const url = max == 'all' ? '/list-comments' : '/list-comments?max-comments=' + max;
+  const max = parseInt(document.getElementById('max-num-comments').value) || Number.MAX_VALUE;
+  const sortOrder = document.getElementById('sort-order').value;
+  let url = '/list-comments?sort-order=' + sortOrder;
   fetch(url)
     .then(response => response.json())
     .then((data) => {
-      const display = document.getElementById('comment-container');
-      display.innerHTML = '';
-      for (let comment of data) {
-        const likeButton = createLikeButton();
-        likeButton.addEventListener('click', () => sendLike(comment));
-
-        const deleteButton = createDeleteButton();
-        deleteButton.addEventListener('click', () => deleteComment(comment));
-
-        const commentElement = createCommentElement(comment);
-        commentElement.appendChild(likeButton);
-        commentElement.appendChild(deleteButton);
-        display.appendChild(commentElement);
-      }
-  });
+      showNextNComments(data, 0, max);
+    });
+}
+
+/**
+ * Display a specified amount of messages starting at a certain index.
+ */
+function showNextNComments(data, startInd, numComments) {
```

Comment:
nit: this might be more readable if numComments was named maxNumComments and data was allComments (doesn't have to be those exactly, but something along those lines).